lib/internal/Magento/Framework/App/Test/Unit/ProductMetadataTest.php

```diff
@@ -40,11 +52,21 @@ protected function setUp()
     public function testGetVersion($packageList, $expectedVersion)
     {
         $this->composerInformationMock->expects($this->any())->method('getSystemPackages')->willReturn($packageList);
+        $this->cacheMock->expects($this->once())->method('save')->with($expectedVersion);
         $productVersion = $this->productMetadata->getVersion();
         $this->assertNotEmpty($productVersion, 'Empty product version');
         $this->assertEquals($expectedVersion, $productVersion);
     }
 
+    public function testGetVersionCached()
```

Comment:
It not looks like caching is tested here, as you're not checking that `save` method was executed for cache mock.

Please update test

Reply:
Updated. Please check now.

Reply:
Seems like you expect that save method should be executed in some case. Am I wrong ?

Reply:
No, "save" method shouldn't be executed if value is loaded from cache. However I added it to `testGetVersion` test, because it's expected there to save the value from composer.

Reply:
In the code you moved code that **saves** data to cache, and as for me - it's correct. 
However in the test you covered another case - when we got some data - save will not be executed (and it wasn't executed even earlier) --> your test isn't covering the case that you've fixed.

Reply:
I covered `save` execution in line 51. Method `testGetVersionCached` doesn't cover the fix indeed, it's just an additional test for loading from cache. Method `testGetVersion` covers the fix in line 51.

Reply:
Ah, sorry that’s my mistake, I missed it and thought you tried to cover it.